### PR TITLE
Add ci

### PR DESCRIPTION
**Pull request by AloeareV**
When adding CI in a PR, it only tests when it's on a branch of the repo it's opened against, as far as I can tell. It's normally nonstandard to push to a branch on the origin, but in the case of CI I think the benefits outweigh the drawbacks.

### CODE REVIEW

**Comment by zancas** [comment]
@dannasessha please see https://github.com/zingolabs/zcashrpc/pull/40
After considering that, please perform a detailed analysis and review of this PR, including:

* skimming GH actions
* evaluating tradeoffs between different login/authentication mechanisms
* walking through the different event triggers

 I am removing myself as a reviewer.

File: .github/workflows/testface.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: cargo-test-quizface
+
+on:
+  pull_request:
+    branches: [ master ]
```
Comment (zancas) [inline .github/workflows/testface.yml:5]: Or..  @AloeareV, is it "main" nowadays?

File: .github/workflows/testface.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    types: [ assigned, opened, synchronize, reopened ]
```
Comment (zancas) [inline .github/workflows/testface.yml:6]: Let's definitely take out "assigned".

File: .github/workflows/testface.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name:  Authenticate Github CR
+      env: 
+        PAT: ${{ secrets.CONTAINER_REGESTRY_READ }}
```
Comment (zancas) [inline .github/workflows/testface.yml:16]: `REGESTRY` ---> `REGISTRY`

File: .github/workflows/testface.yml (hunk 1 above)
Comment (AloeareV) [reply .github/workflows/testface.yml:5]: It's only main for repos created after a certain version or something

File: .github/workflows/testface.yml (hunk 2 above)
Comment (AloeareV) [reply .github/workflows/testface.yml:6]: Sure. Without assigned, it's the default set, so we can drop this line entirely.

**Review by zancas — COMMENTED** [review]
great!

**Pull request merged**